controller_manager:
  ros__parameters:
    update_rate: 90

    # Arm on CAN interface
    velocity_arm_can_controller:   
      type: joint_trajectory_controller/JointTrajectoryController

    # Arm on Serial (USB) interface
    velocity_arm_serial_controller:   
      type: joint_trajectory_controller/JointTrajectoryController

    velocity_gripper_controller:  
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster


# === CAN Controller ===
velocity_arm_can_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity
    gains:
      joint_1: {p: 1.2, i: 0.0, d: 0.00, i_clamp: 0.0}
      joint_2: {p: 1.2, i: 0.0, d: 0.00, i_clamp: 0.0, antiwindup: true}
      joint_3: {p: 1.5, i: 0.0, d: 0.00, i_clamp: 0.0, antiwindup: true}
    open_loop_control: false


# === Serial (ESP) Controller ===
velocity_arm_serial_controller:
  ros__parameters:
    joints:
      - joint_4
      - joint_5
      - joint_6
      - joint_7
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity
    gains:
      joint_4: {p: 1.0, i: 0.0, d: 0.00, i_clamp: 0.0, antiwindup: true}
      joint_5: {p: 1.5, i: 0.0, d: 0.03, i_clamp: 0.0, antiwindup: true}
      joint_6: {p: 1.75, i: 0.0, d: 0.00, i_clamp: 0.0, antiwindup: true}
      joint_7: {p: 2.25, i: 0.0, d: 0.00, i_clamp: 0.0, antiwindup: true}
    open_loop_control: false


# === Gripper Controller ===
velocity_gripper_controller:
  ros__parameters:
    joints:
      - gear1_joint
    command_interfaces: 
      - velocity
    state_interfaces:
      - position
      - velocity
    gains:
      gear1_joint: {p: 1.5, i: 0.0, d: 0.0, angle_wraparound: true}
    open_loop_control: false
    allow_integration_in_goal_trajectories: true


# === Joint State Broadcaster ===
joint_state_broadcaster:
  ros__parameters:
    update_rate: 100
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
      - joint_7
      - gear1_joint
    interfaces:
      - position
      - velocity
      - effort
